lib/cocoapods/installer/xcode/pods_project_generator/file_references_installer.rb

```diff
@@ -203,11 +209,11 @@ def add_file_accessors_paths_to_pods_group(file_accessor_key, group_key = nil, r
               next if paths.empty?
 
               pod_name = file_accessor.spec.name
-              local = sandbox.local?(pod_name)
-              base_path = local ? common_path(paths) : nil
+              preserve_file_structure_flag = (sandbox.local?(pod_name) || preserve_file_structure)
+              base_path = preserve_file_structure_flag ? common_path(paths) : nil
               group = pods_project.group_for_spec(pod_name, group_key)
               paths.each do |path|
-                pods_project.add_file_reference(path, group, local && reflect_file_system_structure_for_development, base_path)
+                pods_project.add_file_reference(path, group, preserve_file_structure_flag && reflect_file_system_structure_for_development, base_path)
```

Comment:
the `reflect_file_system_structure_for_development` function argument could be repurposed here to be `reflect_file_system_structure`, and remove the check for `local`. then the places that call `add_file_accessors_paths_to_pods_group` could reference the `preserve_file_structure` setting. that keeps this function flexible and it would rely less on internal state